src/features/courses/course-route.ts

```diff
@@ -205,17 +202,51 @@ router.get('/questions',
 
         let topic: CourseTopicContent | null = null;
         if(!_.isNil(req.query.courseTopicContentId)) {
-            topic = await courseController.getTopicById(req.query.courseTopicContentId, userId);
+            topic = await courseController.getTopicById({
+                id: req.query.courseTopicContentId,
+                userId,
+            });
             const overrideStartDate = topic.studentTopicOverride?.[0]?.startDate;
             const startDate = overrideStartDate ?? topic.startDate;
 
+            const user = await req.session.getUser();
             if (moment().isBefore(startDate)) {
-                const user = await req.session.getUser();
                 if (user.roleId === Role.STUDENT) {
                     next(Boom.badRequest(`The topic "${topic.name}" has not started yet.`));
                     return;
                 }
             }
+
+            if (topic?.topicTypeId === 2 && !_.isNil(userId)) {
+                if (_.isNil(topic.topicAssessmentInfo)){
+                    next(Boom.badRequest('Topic is an assessment, but does not have corresponding assessment info. This should never happen.'));
+                    return;
+                }
+
+                // get version info - descending by startTime unless specific id is included in query
+                let versions: StudentTopicAssessmentInfo[] | undefined;
+                if (_.isNil(req.query.studentTopicAssessmentInfoId)) {
+                    versions = _.orderBy(topic.topicAssessmentInfo.studentTopicAssessmentInfo, ['startTime'],'desc');
```

Comment:
If the other one wasn't working, this one might also need a fix
```suggestion
                    versions = _.orderBy(topic.topicAssessmentInfo.studentTopicAssessmentInfo, ['startTime'], ['desc']);
```

Reply:
already fixed in the next branch